src/quic/defs.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include <env.h>
+#include <node_errors.h>
+#include <v8.h>
+
+namespace node {
+namespace quic {
+
+template <typename Opt, bool Opt::*member>
+bool SetOption(Environment* env,
+               Opt* options,
+               const v8::Local<v8::Object>& object,
+               const v8::Local<v8::String>& name) {
+  v8::Local<v8::Value> value;
+  if (!object->Get(env->context(), name).ToLocal(&value)) return false;
+  if (!value->IsUndefined()) {
+    CHECK(value->IsBoolean());
```

Comment:
Do we need this? At least I recommend `DCHECK`

Reply:
Given that this is interacting with user-provided data that *should* be type checked at the JS layer but *could* potentially be monkeypatched, I think having the explicit `CHECK` rather than a `DCHECK` here makes sense.